geopyspark/protobufregistry.py

```diff
@@ -0,0 +1,455 @@
+        elif data_type == "FLOAT":
+            ctype = tile.cellType
+            ctype.dataType = ProtoCellType.FLOAT
+            tile.floatCells.extend(arr.flatten().tolist())
+        else:
+            cell_type.dataType = ProtoCellType.DOUBLE
+            tile.doubleCells.extend(arr.flatten().tolist())
+
+        return tile
+
+    @classmethod
+    def tile_encoder(cls, obj):
+        """Decodes a ``TILE`` into bytes.
+
+        Args:
+            obj (:ref:`Tile <raster>`): An instance of ``Extent``.
+
+        Returns:
+            bytes
+        """
+
+        return cls._tile_encoder(obj).SerializeToString()
+
+    @classmethod
+    def _multibandtile_encoder(cls, obj):
+        if obj['data'].ndim == 2:
+            obj['data'] = np.expand_dims(obj['data'], 0)
+
+        band_count = obj['data'].shape[0]
+
+        def create_dict(index):
+            return {'data': obj['data'][index, :, :], 'no_data_value': obj['no_data_value'],
+                    'data_type': obj['data_type']}
+
+        multibandtile = ProtoMultibandTile()
+        multibandtile.tiles.extend([cls._tile_encoder(create_dict(x)) for x in range(band_count)])
+
+        return multibandtile
+
+    @classmethod
+    def multibandtile_encoder(cls, obj):
+        """Decodes a ``TILE`` into bytes.
+
+        Args:
+            obj (:ref:`Tile <raster>`): An instance of ``Extent``.
+
+        Returns:
+            bytes
+        """
+
+        return cls._multibandtile_encoder(obj).SerializeToString()
+
+    @staticmethod
+    def extent_encoder(obj):
+        """Encodes an ``Extent`` into bytes.
+
+        Args:
+            obj (:class:`~geopyspark.geotrellis.Extent`): An instance of ``Extent``.
+
+        Returns:
+            bytes
+        """
+
+        return obj.to_protobuf_extent.SerializeToString()
+
+    @staticmethod
+    def projected_extent_encoder(obj):
+        """Encodes a ``ProjectedExtent`` into bytes.
+
+        Args:
+            obj (:class:`~geopyspark.geotrellis.ProjectedExtent`): An instance of
+                ``ProjectedExtent``.
+
+        Returns:
+            bytes
+        """
+
+        return obj.to_protobuf_projected_extent.SerializeToString()
+
+    @staticmethod
+    def temporal_projected_extent_encoder(obj):
+        """Encodes a ``TemproalProjectedExtent`` into bytes.
+
+        Args:
+            obj (:class:`~geopyspark.geotrellis.TemporalProjectedExtent`): An instance of
+                ``TemporalProjectedExtent``.
+
+        Returns:
+            bytes
+        """
+
+        return obj.to_protobuf_temporal_projected_extent.SerializeToString()
+
+    @staticmethod
+    def spatial_key_encoder(obj):
+        """Encodes a ``SpatialKey`` into bytes.
+
+        Args:
+            obj (:class:`~geopyspark.geotrellis.SpatialKey`): An instance of ``SpatialKey``.
+
+        Returns:
+            bytes
+        """
+
+        return obj.to_protobuf_spatial_key.SerializeToString()
```

Comment:
this method should just be inlined here and avoid making named tuples into classes.